### DISCUSSION TITLE
behavior for .zwc files

### DISCUSSION THREAD

**Discussion by drakeaxelrod** [marlonrichert/zsh-snap#70] [Ideas]
I don't know if its the the intended behavior of znap to put all the .zwc files in whatever dir the standard file is located. if it is not then I probably messed up the install trying to keep my settings etc. If it is the intended behavior I would suggest maybe a folder inside the zsh-snap folder that stores these files, simply so that if like me you split up your zsh files you dont have duplicate .zwc files all over your zsh dir.

**Comment by marlonrichert** [comment]
> I don't know if its the the intended behavior of znap to put all the .zwc files in whatever dir the standard file is located.

Yes, that is the intended behavior.

> I would suggest maybe a folder inside the zsh-snap folder that stores these files

That's unfortunately not possible. For almost all cases, the compiled `.zwc` file needs to be in the same dir as the source file. (The only exception is functions loaded through the `$fpath`. For those, you can compile a digest of all functions in an `$fpath` dir, but even then, the digest needs to be located in the same parent dir.)

> if like me you split up your zsh files you dont have duplicate .zwc files all over your zsh dir.

If you don't want `.zwc` files to be generated for your dotfiles, you can exclude them by adding something like this to your `.zshrc` file:
```zsh
zstyle ':znap:*' auto-compile-ignore "${ZDOTDIR:-$HOME}/.z*"
```
This is also explained [in the Readme](https://github.com/marlonrichert/zsh-snap#asynchronous-compilation).
